### PR TITLE
Use symbols for build status

### PR DESCRIPTION
**Pull request by janfaracik**
Rather late followup to #7208 in which weather icons were replaced with [symbols](https://weekly.ci.jenkins.io/design-library/Symbols/). This PR does the same for build status icons and also updates the appearance of some of them (do give feedback on this!).

The current approach to build status icons is a little unwieldy and can break in some scenarios (see below images). Using symbols is consistent with near enough every other icon in core Jenkins and keeps things simple to update in the future.

**Before**
image

**After**
image

**Before (in Safari)**
image

**After**
image

### Testing done

* Icons appear as expected on the dashboard
* Icons appear as expected in build history (page and widget format)
* Icons appear as expected in build time trend (fixes a UI issue in Safari too)

### Proposed changelog entries

- Refine build status icons

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@jenkinsci/sig-ux 

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/resources/hudson/model/Job/buildTimeTrend.jelly
Comment (mawinter69) [inline core/src/main/resources/hudson/model/Job/buildTimeTrend.jelly:34]: That id seems to be rather generic and bears the risk of collision. I would suggest to apply here the rules as defined in the design-library and prefix with `jenkins-` or `app-`

File: core/src/main/resources/lib/hudson/buildListTable.jelly
Comment (mawinter69) [inline core/src/main/resources/lib/hudson/buildListTable.jelly:33]: same as above

**Review by mawinter69 — CHANGES_REQUESTED** [review]

File: core/src/main/resources/hudson/model/Job/buildTimeTrend.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/model/Job/buildTimeTrend.jelly]: Done.

File: core/src/main/resources/lib/hudson/buildListTable.jelly
Comment (janfaracik) [reply core/src/main/resources/lib/hudson/buildListTable.jelly]: Done.

**Comment by timja** [comment]
The alignment looks a little off in the build history?

Before:
image


After:

image

----

Apart from that looks better to me

**Comment by janfaracik** [comment]
> The alignment looks a little off in the build history?
> 
> Before: image
> 
> After:
> 
> image
> Apart from that looks better to me

Not seeing that on my end?

`master`
image

`use-symbols-for-build-status`
image

**Comment by timja** [comment]
I see it in your screenshot to, maybe not required to fix it's just a bit higher than before.

**Comment by janfaracik** [comment]
> I see it in your screenshot to, maybe not required to fix it's just a bit higher than before.

Ah sorry I misunderstood - the icon position is fixed now.

**Review by timja — APPROVED** [review]
LGTM

**Review by NotMyFault — APPROVED** [review]
Looks great!

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by mawinter69 — CHANGES_REQUESTED** [review]
I have to object. There are still several places where the old things are used. I even found a place where an old blue ball is used.
E.g. create a freestyle job that triggers another freestyle job. This lead to this:
image
Maybe that old blue ball is coming from https://plugins.jenkins.io/parameterized-trigger/

Places with old implementation:
lib/hudson/buildLink.jelly#55
hudon/model/AbstractBuild/index.jelly#79,#89
hudson/model/Run/statusIcon.jelly#34 (not sure if this is anywhere used
lib/hudson/jobLink.jelly#34

**Comment by mawinter69** [comment]
Seems the blue ball was recently fixed: https://github.com/jenkinsci/parameterized-trigger-plugin/pull/357

**Comment by janfaracik** [comment]
> I have to object. There are still several places where the old things are used. I even found a place where an old blue ball is used. E.g. create a freestyle job that triggers another freestyle job. This lead to this: image Maybe that old blue ball is coming from https://plugins.jenkins.io/parameterized-trigger/
> 
> Places with old implementation: lib/hudson/buildLink.jelly#55 hudon/model/AbstractBuild/index.jelly#79,#89 hudson/model/Run/statusIcon.jelly#34 (not sure if this is anywhere used lib/hudson/jobLink.jelly#34

Updated those - thanks for finding.

**Review by mawinter69 — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by MarkEWaite** [comment]
@janfaracik thanks very much for the improvement in the build status display!  I like it very much.

It appears that the improvement has exposed two unit tests in the JUnit plugin that were dependent on the previous layout.  Would you be willing to look at those two tests to adapt them so that they will pass on both Jenkins 2.433 (before the change) and Jenkins 2.434 (after the change)?

Detected in plugin bill of materials pull request:

* https://github.com/jenkinsci/bom/pull/2704

Raised as a JUnit plugin issue in:

* https://github.com/jenkinsci/junit-plugin/issues/588

File: core/src/main/resources/hudson/model/Job/buildTimeTrend_resources.js
Comment (mawinter69) [reply core/src/main/resources/hudson/model/Job/buildTimeTrend_resources.js:102]: The problem is not here but in StatusColumn -> column.jelly

File: core/src/main/resources/hudson/model/Job/buildTimeTrend_resources.js
Comment (mawinter69) [inline core/src/main/resources/hudson/model/Job/buildTimeTrend_resources.js:72]: any reason why the anime svgs are excluded? Before they were animated when the build was running

File: core/src/main/resources/lib/hudson/buildLink.jelly
Comment (uhafner) [inline core/src/main/resources/lib/hudson/buildLink.jelly:55]: This breaks the build link again (3 time in the last past months) that is shown in the warnings and coverage links. Maybe we should provide a UI tests for this feature.

Bildschirmfoto 2023-12-23 um 17 16 32

**Comment by uhafner** [comment]
FTR: I filed https://issues.jenkins.io/browse/JENKINS-72491

**Comment by daniel-beck** [comment]
Caused [JENKINS-72711](https://issues.jenkins.io/browse/JENKINS-72711)

File: core/src/main/resources/hudson/model/Job/buildTimeTrend_resources.js
Comment (daniel-beck) [reply core/src/main/resources/hudson/model/Job/buildTimeTrend_resources.js:72]: This is a fun unaddressed comment to find after triaging https://github.com/jenkinsci/jenkins/pull/8705#issuecomment-1943677731

**Comment by daniel-beck** [comment]
Caused [JENKINS-72845](https://issues.jenkins.io/browse/JENKINS-72845).

**Comment by alexsch01** [comment]
I made https://issues.jenkins.io/browse/JENKINS-73386
TLDR: I believe the aborted button and disabled button look too similar due to this PR

**Comment by daniel-beck** [comment]
Caused [JENKINS-74868](https://issues.jenkins.io/browse/JENKINS-74868).

**Pull request merged**
